### PR TITLE
[ntuple] Return unique pointer from RFieldBase::Create and Clone

### PR DESCRIPTION
**Pull request by jblomer**
Includes #6643

### CODE REVIEW

File: tree/ntuple/v7/inc/ROOT/RField.hxx [hunk 1]
```diff
@@ -472,7 +472,9 @@ public:
    RField(RField&& other) = default;
    RField& operator =(RField&& other) = default;
    ~RField() = default;
-   RFieldBase* Clone(std::string_view newName) final { return new RField(newName); }
+   std::unique_ptr<Detail::RFieldBase> Clone(std::string_view newName) const final {
+      return std::unique_ptr<Detail::RFieldBase>(new RField(newName));
```
Comment (pcanal) [inline tree/ntuple/v7/inc/ROOT/RField.hxx:476]:
```suggestion
      return std::make_unique<Detail::RField>(newName);
```

File: tree/ntuple/v7/src/RNTupleModel.cxx [hunk 2]
```diff
@@ -43,9 +43,9 @@ ROOT::Experimental::RNTupleModel::RNTupleModel()
 ROOT::Experimental::RNTupleModel* ROOT::Experimental::RNTupleModel::Clone()
```
Comment (pcanal) [inline tree/ntuple/v7/src/RNTupleModel.cxx:43]: Is the an advantage to having Clone being a non-const function?

File: tree/ntuple/v7/src/RNTupleModel.cxx [hunk 3]
```diff
@@ -43,9 +43,9 @@ ROOT::Experimental::RNTupleModel::RNTupleModel()
...
 {
    auto cloneModel = new RNTupleModel();
-   auto cloneFieldZero = static_cast<RFieldZero*>(fFieldZero->Clone(""));
-   cloneModel->fFieldZero = std::unique_ptr<RFieldZero>(cloneFieldZero);
-   cloneModel->fDefaultEntry = std::unique_ptr<REntry>(cloneFieldZero->GenerateEntry());
+   auto cloneFieldZero = fFieldZero->Clone("");
+   cloneModel->fFieldZero = std::unique_ptr<RFieldZero>(static_cast<RFieldZero *>(cloneFieldZero.release()));
+   cloneModel->fDefaultEntry = std::unique_ptr<REntry>(cloneModel->fFieldZero->GenerateEntry());
```
Comment (pcanal) [inline tree/ntuple/v7/src/RNTupleModel.cxx:48]: Shouldn't GenerateEntry also return a std::unique_ptr and then use the same pattern as the line before (which ever we settle on)?

Comment (jblomer) [reply tree/ntuple/v7/src/RNTupleModel.cxx:48]: Yep, addressed in #6655

File: tree/ntuple/v7/src/RNTupleModel.cxx (hunk 2 above)
Comment (jblomer) [reply tree/ntuple/v7/src/RNTupleModel.cxx:43]: Addressed in #6655

**Review by pcanal — APPROVED** [review]

**Comment by jblomer** [comment]
@phsft-bot build on on mac1015/cxx17

**Comment by jblomer** [comment]
@phsft-bot build on mac1015/cxx17

**Pull request merged**